[bug] Using CMakeToolchain + msvc impossible with standard installation of VS2015

### Environment Details
  * Operating System+version: Windows ANY
  * Compiler+version: `cl.exe 19.00` (`msvc 19.0` according to `settings.yml`) from a standard VS2015 installation
  * Conan version: 1.41.0
  * Python version: 3.8.9

### Steps to reproduce

When using the `msvc 19.0` compiler of a "standard" Visual Studio 2015 installation ("standard" meaning that it's not the "MSVC v140 - VS 2015 C++ Build Tools (v14.00)" included in later editions VS2017 & VS2019), the `CMakeToolchain` crashes while calling the generated `conanvcvars.bat`. By examining the latter (see below), it turns out the argument `-vcvars_ver` is passed to the `vcvarsall.bat` activation script. However, that particular version of the script doesn't expect such an argument - since it was added from VS2017 onwards for activating older toolsets in newer versions of Visual Studio - so it errors out.

```batch
@echo off
set "VSCMD_START_DIR=%CD%" && call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat"  amd64 -vcvars_ver=14.0
```

 whereas the correct call should be:

```batch
@echo off
set "VSCMD_START_DIR=%CD%" && call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat"  amd64
```

### Logs

```
Configuration (profile_host):
[settings]
arch=x86_64
build_type=Debug
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=19.0
os=Windows
[options]
*:shared=True
[build_requires]
[env]
CC=cl
CXX=cl
Configuration (profile_build):
[settings]
arch=x86_64
build_type=Debug
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=19.0
os=Windows
[options]
[build_requires]
[env]
CC=cl
CXX=cl
zlib/1.2.11@compression/stable: Forced build from source
zlib/1.2.11@compression/stable (test package): Installing package
Requirements
    zlib/1.2.11@compression/stable from local cache - Cache
Packages
    zlib/1.2.11@compression/stable:75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3 - Build

Installing (downloading, building) binaries...
zlib/1.2.11@compression/stable: Configuring sources in C:\gitlab-runner\builds\9fb8a5b5\0\io\zlib\.conan\data\zlib\1.2.11\compression\stable\source

zlib/1.2.11@compression/stable: Copying sources to build folder
zlib/1.2.11@compression/stable: Building your package in C:\gitlab-runner\builds\9fb8a5b5\0\io\zlib\.conan\data\zlib\1.2.11\compression\stable\build\75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3
zlib/1.2.11@compression/stable: Generator txt created conanbuildinfo.txt
zlib/1.2.11@compression/stable: Calling generate()
zlib/1.2.11@compression/stable: Aggregating env generators
zlib/1.2.11@compression/stable: Calling build()
zlib/1.2.11@compression/stable: CMake command: cmake -G "Ninja" -DCMAKE_TOOLCHAIN_FILE="C:/gitlab-runner/builds/9fb8a5b5/0/io/zlib/.conan/data/zlib/1.2.11/compression/stable/build/75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/gitlab-runner/builds/9fb8a5b5/0/io/zlib/.conan/data/zlib/1.2.11/compression/stable/package/75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3" "C:\gitlab-runner\builds\9fb8a5b5\0\io\zlib\.conan\data\zlib\1.2.11\compression\stable\build\75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3\_source_subfolder"

----Running------
> "C:\gitlab-runner\builds\9fb8a5b5\0\io\zlib\.conan\data\zlib\1.2.11\compression\stable\build\75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3\conanbuild.bat" && cmake -G "Ninja" -DCMAKE_TOOLCHAIN_FILE="C:/gitlab-runner/builds/9fb8a5b5/0/io/zlib/.conan/data/zlib/1.2.11/compression/stable/build/75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/gitlab-runner/builds/9fb8a5b5/0/io/zlib/.conan/data/zlib/1.2.11/compression/stable/package/75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3" "C:\gitlab-runner\builds\9fb8a5b5\0\io\zlib\.conan\data\zlib\1.2.11\compression\stable\build\75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3\_source_subfolder"
-----------------
Logging command output to file 'C:\gitlab-runner\builds\9fb8a5b5\0\io\zlib\.conan\data\zlib\1.2.11\compression\stable\build\75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3\_build_subfolder\conan_run.log'
Error in script usage. The correct usage is:

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" [option]

  or

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" [option] store

  or

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" [option] [version number]

  or

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" [option] store [version number]

where [option] is: x86 | amd64 | arm | x86_amd64 | x86_arm | amd64_x86 | amd64_arm

where [version number] is either the full Windows 10 SDK version number or "8.1" to use the windows 8.1 SDK

:

The store parameter sets environment variables to support

  store (rather than desktop) development.

:

For example:

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" x86_amd64

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" x86_arm store

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" x86_amd64 10.0.10240.0

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" x86_arm store 10.0.10240.0

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" x64 8.1

    "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC/vcvarsall.bat" x64 store 8.1

:

Please make sure either Visual Studio or C++ Build SKU is installed.

Using Conan toolchain through C:/gitlab-runner/builds/9fb8a5b5/0/io/zlib/.conan/data/zlib/1.2.11/compression/stable/build/75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3/conan_toolchain.cmake.
-- Conan toolchain: Setting BUILD_SHARED_LIBS= ON
-- The C compiler identification is unknown
CMake Error at CMakeLists.txt:4 (project):
  The CMAKE_C_COMPILER:

    cl

  is not a full path and was not found in the PATH.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "C:/gitlab-runner/builds/9fb8a5b5/0/io/zlib/.conan/data/zlib/1.2.11/compression/stable/build/75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3/_build_subfolder/CMakeFiles/CMakeOutput.log".
See also "C:/gitlab-runner/builds/9fb8a5b5/0/io/zlib/.conan/data/zlib/1.2.11/compression/stable/build/75d5ce8d183d5e36dab24b5c22b2e9d17f3981e3/_build_subfolder/CMakeFiles/CMakeError.log".
```

On a different build node equipped with VS2019 Build Tools + MSVC v140 toolset and using the same profile with the sole addition of `tools.microsoft.msbuild:vs_version=16`, it worked just fine.

However, I would expect conan to do the right thing (stopping short of specifying `-vcvars_ver`) also when activating a "traditional" VS2015 installation.

--- a/conan/tools/microsoft/visual.py
+++ b/conan/tools/microsoft/visual.py
@@ -26,23 +26,8 @@ class VCVars:
 
         vs_version = vs_ide_version(conanfile)
         vcvarsarch = vcvars_arch(conanfile)
-        vcvars_ver = None
-        if compiler == "Visual Studio":
-            toolset = conanfile.settings.get_safe("compiler.toolset")
-            if toolset is not None:
-                vcvars_ver = {"v140": "14.0",
-                              "v141": "14.1",
-                              "v142": "14.2",
-                              "v143": "14.3"}.get(toolset)
-        else:
-            assert compiler == "msvc"
-            # Code similar to CMakeToolchain toolset one
-            compiler_version = str(conanfile.settings.compiler.version)
-            version_components = compiler_version.split(".")
-            assert len(version_components) >= 2  # there is a 19.XX
-            minor = version_components[1]
-            # The equivalent of compiler 19.26 is toolset 14.26
-            vcvars_ver = "14.{}".format(minor)
+        vcvars_ver = _vcvars_vers(conanfile, compiler, vs_version)
+
         vs_install_path = conanfile.conf["tools.microsoft.msbuild:installation_path"]
         # The vs_install_path is like
         # C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
@@ -163,3 +148,27 @@ def vcvars_arch(conanfile):
         raise ConanException('vcvars unsupported architectures %s-%s' % (arch_build, arch_host))
 
     return arch
+
+
+def _vcvars_vers(conanfile, compiler, vs_version):
+    if int(vs_version) <= 14:
+        return None
+
+    vcvars_ver = None
+    if compiler == "Visual Studio":
+        toolset = conanfile.settings.get_safe("compiler.toolset")
+        if toolset is not None:
+            vcvars_ver = {"v140": "14.0",
+                          "v141": "14.1",
+                          "v142": "14.2",
+                          "v143": "14.3"}.get(toolset)
+    else:
+        assert compiler == "msvc"
+        # Code similar to CMakeToolchain toolset one
+        compiler_version = str(conanfile.settings.compiler.version)
+        version_components = compiler_version.split(".")
+        assert len(version_components) >= 2  # there is a 19.XX
+        minor = version_components[1]
+        # The equivalent of compiler 19.26 is toolset 14.26
+        vcvars_ver = "14.{}".format(minor)
+    return vcvars_ver

--- a/conans/test/integration/toolchains/microsoft/vcvars_test.py
+++ b/conans/test/integration/toolchains/microsoft/vcvars_test.py
@@ -51,3 +51,22 @@ def test_vcvars_generator_string():
                '-s compiler.cppstd=14 -s compiler.runtime=static')
 
     assert os.path.exists(os.path.join(client.current_folder, "conanvcvars.bat"))
+
+
+@pytest.mark.skipif(platform.system() != "Windows", reason="Requires Windows")
+def test_vcvars_2015_error():
+    # https://github.com/conan-io/conan/issues/9888
+    client = TestClient(path_with_spaces=False)
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+        class TestConan(ConanFile):
+            generators = "VCVars"
+            settings = "os", "compiler", "arch", "build_type"
+    """)
+    client.save({"conanfile.py": conanfile})
+    client.run('install . -s os=Windows -s compiler="msvc" -s compiler.version=19.0 '
+               '-s compiler.cppstd=14 -s compiler.runtime=static')
+
+    vcvars = client.load("conanvcvars.bat")
+    assert "-vcvars_ver" not in vcvars

--- a/conans/test/unittests/tools/cmake/test_cmaketoolchain.py
+++ b/conans/test/unittests/tools/cmake/test_cmaketoolchain.py
@@ -167,6 +167,7 @@ def test_osx_deployment_target(conanfile_apple):
     content = toolchain.content
     assert 'set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "")' in content
 
+
 @pytest.fixture
 def conanfile_msvc():
     c = ConanFile(Mock(), None)
@@ -186,6 +187,7 @@ def conanfile_msvc():
     c._conan_node = Mock()
     c._conan_node.dependencies = []
     return c
+
 
 def test_toolset(conanfile_msvc):
     toolchain = CMakeToolchain(conanfile_msvc)

